quicktest.go

```diff
@@ -83,19 +86,39 @@ func (c *C) Run(name string, f func(c *C)) bool {
 func check(fail func(...interface{}), checker Checker, got interface{}, args []interface{}) bool {
 	// Ensure that we have a checker.
 	if checker == nil {
-		fail(report("cannot run test: nil checker provided"))
+		fail(report(testerror.BadCheckf("cannot run test: nil checker provided"), nil))
 		return false
 	}
-	// Validate that we have the correct number of arguments.
+	// Extract a comment if it has been provided.
 	gotNumArgs, wantNumArgs := len(args), checker.NumArgs()
```

Comment:
tbh I'm not sure it's helpful to declare gotNumArgs separately. I think I'd probably just use len(args) everywhere, then you wouldn't need to keep gotNumArgs in sync and the code's arguably more obvious.

Reply:
Done.